blob/src/main/java/io/crate/blob/BlobContainer.java

```diff
@@ -91,21 +87,22 @@ private void createSubDirectories(File parentDir) {
         }
     }
 
-    public interface FileVisitor {
-
-        boolean visit(File file) throws IOException;
-
+    public Iterable<File> getFiles() {
+        final Collection<File> files = new LinkedList<>();
```

Comment:
That's exactly the reason why a custom Iterable should be implemented - to avoid copying stuff around